<launch>
    <arg name="urdf_file" default="$(find jaco_description)/urdf/jaco_robot.urdf"/>
    
    <!-- command line parameter -->
    <arg name="iv_or_link" default="jaco_6_hand_limb"/>
    
    <arg name="join_fixed_links" default="true"/>
    <arg name="rotate_axes_z" default="true"/>
    <arg name="display_axes" default="true"/>
    <arg name="axes_radius" default="0.001"/>
    <arg name="axes_length" default="0.015"/>

    # An axis and angle (degrees) can be specified which will transform *all*
    # visuals (not links, but their visuals!) within their local coordinate system.
    # This can be used to correct transformation errors which may have been 
    # introduced in converting meshes from one format to the other, losing orientation information
    # For example, .dae has an "up vector" definition which may have been ignored.
    <arg name="visual_corr_axis_x" default="1"/>
    <arg name="visual_corr_axis_y" default="0"/>
    <arg name="visual_corr_axis_z" default="0"/>
    <arg name="visual_corr_axis_angle" default="0"/>

	<node name="urdf_viewer" pkg="urdf_viewer" type="urdf_viewer_node" respawn="false"
        output="screen" args="$(arg urdf_file) $(arg iv_or_link)">
    <param name="join_fixed_links" value="$(arg join_fixed_links)"/>
    <param name="rotate_axes_z" value="$(arg rotate_axes_z)"/>
    <param name="display_axes" value="$(arg display_axes)"/>
    <param name="axes_radius" value="$(arg axes_radius)"/>
    <param name="axes_length" value="$(arg axes_length)"/>
    <param name="visual_corr_axis_x" value="$(arg visual_corr_axis_x)"/>
    <param name="visual_corr_axis_y" value="$(arg visual_corr_axis_y)"/>
    <param name="visual_corr_axis_z" value="$(arg visual_corr_axis_z)"/>
    <param name="visual_corr_axis_angle" value="$(arg visual_corr_axis_angle)"/>
  </node>
</launch>
